pandas/_libs/join.pyi

```diff
@@ -55,39 +55,39 @@ def asof_join_backward_on_X_by_Y(
     left_by_values: np.ndarray,  # by_t[:]
     right_by_values: np.ndarray,  # by_t[:]
     allow_exact_matches: bool = ...,
-    tolerance=...,
+    tolerance: np.number | int | float | None = ...,
```

Comment:
umm these can be a timedelta as well

Reply:
im not sure thats correct.  in merge.py we do
```
        if needs_i8_conversion(left_values):
            left_values = left_values.view("i8")
            right_values = right_values.view("i8")
            if tolerance is not None:
                tolerance = Timedelta(tolerance)
                tolerance = tolerance.value
```

so we'd get a int64

Reply:
oh that's a good point